third-party/CMakeLists.txt

```diff
@@ -144,14 +144,8 @@ endif()
 
 include(ExternalProject)
 
-if(WIN32)
-  # "nvim" branch of https://github.com/neovim/libuv
-  set(LIBUV_URL https://github.com/neovim/libuv/archive/b899d12b0d56d217f31222da83f8c398355b69ef.tar.gz)
-  set(LIBUV_SHA256 eb7e37b824887e1b31a4e31d1d9bad4c03d8b98532d9cce5f67a3b70495a4b2a)
-else()
-  set(LIBUV_URL https://github.com/libuv/libuv/archive/v1.42.0.tar.gz)
-  set(LIBUV_SHA256 371e5419708f6aaeb8656671f89400b92a9bba6443369af1bb70bcd6e4b3c764)
-endif()
+set(LIBUV_URL https://github.com/libuv/libuv/archive/v1.42.0.tar.gz)
+set(LIBUV_SHA256 371e5419708f6aaeb8656671f89400b92a9bba6443369af1bb70bcd6e4b3c764)
```

Comment:
oh wow, if we can eliminate our fork that will be wonderful

Reply:
> oh wow, if we can eliminate our fork that will be wonderful.

Yes, that was the hope. 

> so rather than deleting if(WIN32) part, we would just update the pinned commit

Or we could wait for the MingW change? (He asks, hopefully?)

Reply:
the MingW doesn't change the fact that there missing tty.c changes. Am I missing something? @erw7 we still need these changes, right? https://github.com/libuv/libuv/compare/v1.x...neovim:nvim

Reply:
(Just to be clear, this was a low-effort test to see whether LibUV 1.42.0 by some chance builds cleanly on Win32 -- this is obviously not the case. Since I don't have access to nor know anything about Windows, I'm not the person to put in any real effort to make this work. Since there's a discussion going on, I'm not simply closing this as I had initially intended, but feel free to do so at any time.)

Reply:
sure. can you try updating to the latest commit on the `nvim` branch of our fork?

Reply:
Sure; let's see if I guessed the hash correctly...

Reply:
> so rather than deleting if(WIN32) part, we would just update the pinned commit .

So the updated commit also pulls in the breaking commit. Did you rebase on the libuv branch or did you cherry-pick all but the commit mentioned in the linked PR?

Do you want to try backing out https://github.com/libuv/libuv/commit/95f88f47f28b4686a6959377c8d2c31daa40bd64 and trying again?

Reply:
> do you want to try backing out [libuv/libuv@95f88f4](https://github.com/libuv/libuv/commit/95f88f47f28b4686a6959377c8d2c31daa40bd64) and trying again?

either that,  or add a fix to our fork until upstream fixes it

Reply:
Backing out the commit seemed to break other things: https://github.com/neovim/neovim/pull/15756/checks?check_run_id=3707501019#step:4:14211

If a Windows expert has an idea, I'd be happy to try it; otherwise I'll probably just close this PR as it doesn't look salvageable.

Reply:
with https://github.com/neovim/neovim/pull/15756/commits/866ff7c924584cb51149e496bf5e9aa682b0e479 i only see one `expect_twostreams` failure, on the MINGW_32 CI, and this is a known flakey test.

Reply:
Well, MINGW_32 is the only test that's affected by that change, so I'm hesitant... But if you say it's safe, I could push those changes to the neovim fork and bump the commit.

(I've re-triggered CI; let's see what happens.)

Reply:
The reason why mignw32 fails to build libuv is due to incomplete modification of libuv/libuv/issues/3178 for MinGW-w64. I don't know if it is correct to report to msys2/MINGW-packages, but I created msys2/MINGW-packages/issues/9667 anyway. I think you should wait until it is fixed on the msys2/MINGW-packages side.

Reply:
`headers-git` just got bumped; I'm trying this again on a new PR.